SYCL/ESIMD/regression/half_conversion_test.cpp

```diff
@@ -0,0 +1,97 @@
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//==- half_conversion_test.cpp - Test for half conversion under ESIMD_EMULATOR
+// backend -==/
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <stdlib.h>
+#include <sycl/ext/intel/esimd.hpp>
+
+#include <iostream>
+
+using namespace ::sycl;
+using namespace ::sycl::ext;
+using namespace sycl::ext::intel;
+using namespace sycl::ext::intel::esimd;
+
+template <typename Ty> struct test_id;
+template <int N>
+using int_type_t = std::conditional_t<
+    N == 1, int8_t,
+    std::conditional_t<
+        N == 2, int16_t,
+        std::conditional_t<N == 4, int32_t,
+                           std::conditional_t<N == 8, int64_t, void>>>>;
+
+template <class Ty> bool test(queue q, int inc) {
+  Ty *data = new Ty[1];
+
+  data[0] = (Ty)0;
+  Ty VAL = (Ty)inc;
+
+  try {
+    buffer<Ty, 1> buf(data, range<1>(1));
+    q.submit([&](handler &cgh) {
+      std::cout << "Running on "
+                << q.get_device().get_info<::sycl::info::device::name>()
+                << "\n";
+      auto acc = buf.template get_access<access::mode::read_write>(cgh);
+      cgh.single_task<test_id<Ty>>([=]() SYCL_ESIMD_KERNEL {
```

Comment:
non-blocking comment. This should work properly without explicit name set:
```suggestion
      cgh.single_task([=]() SYCL_ESIMD_KERNEL {
```